api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/CommonQueries.java

```diff
@@ -889,4 +889,79 @@ public String InitialArtStartDateOverallQuery(String endDate, Integer location)
 
     return stringSubstitutor.replace(sql);
   }
+
+  /**
+   * AND whose first ever drug pick-up date between the following sources falls during the reporting
+   * period:
+   *
+   * <ul>
+   *   <li>Drug pick-up date registered on (FILA)
+   *   <li>Drug pick-up date registered on (Recepção Levantou ARV) – Master Card
+   * </ul>
+   *
+   * @return String
+   */
+  public String getFirstDrugPickup() {
+
+    return "SELECT first.patient_id, "
+        + "       first.first_pickup "
+        + "FROM   (SELECT p.patient_id, "
+        + "               first_ever.pickup_date AS first_pickup "
+        + "        FROM   patient p "
+        + "               INNER JOIN encounter e "
+        + "                       ON e.patient_id = p.patient_id "
+        + "               INNER JOIN (SELECT p.patient_id, "
+        + "                                  Min(e.encounter_datetime) AS pickup_date "
+        + "                           FROM   patient p "
+        + "                                  INNER JOIN encounter e "
+        + "                                          ON e.patient_id = p.patient_id "
+        + "                           WHERE  e.encounter_type = ${18} "
+        + "                                  AND e.voided = 0 "
+        + "                                  AND p.voided = 0 "
+        + "                                  AND e.location_id = :location "
+        + "                                  AND e.encounter_datetime <= :endDate "
+        + "                           GROUP  BY p.patient_id) first_ever "
+        + "                       ON first_ever.patient_id = p.patient_id "
+        + "        WHERE  p.voided = 0 "
+        + "               AND e.voided = 0 "
+        + "               AND e.encounter_type = ${18} "
+        + "               AND e.location_id = :location "
+        + "               AND e.encounter_datetime BETWEEN :startDate AND :endDate "
+        + "        GROUP  BY p.patient_id "
```

Comment:
This query is too complicated for what is intended

Reply:
Resolved